packages/react/src/components/badge/Badge.tsx

```diff
@@ -24,35 +58,42 @@ interface BadgeWithIconProps extends BadgeBasicProps {
      * Add an icon to the badge (Required if no label)
      */
     icon: SvgName;
+    /**
+     * Whether the icon is left or right (Required if no label)
+     */
+    iconPlacement: BadgeIconPlacement;
 }
-
 export type IBadgeProps = BadgeWithLabelProps | BadgeWithIconProps | (BadgeWithLabelProps & BadgeWithIconProps);
 
 export class Badge extends React.Component<IBadgeProps> {
-    static defaultProps: Partial<IBadgeProps> = {
+    static defaultProps: IBadgeProps = {
         extraClasses: [],
+        type: BadgeType.Default,
+        isSmall: false,
+        icon: undefined,
+        iconPlacement: undefined,
     };

```

Comment:
Why do you set the defaultProps to undefined instead of just not defining a default value?

```suggestion
    static defaultProps: Partial<IBadgeProps> = {
        extraClasses: [],
        type: BadgeType.Default,
        isSmall: false,
        iconPlacement: BadgeIconPlacement.Left,
    };
```

Reply:
I've put `label: ' '` instead. I need icon or label otherwise the code throws errors 😕